src/battle_util.c

```diff
@@ -4794,6 +4801,18 @@ u32 AbilityBattleEffects(u32 caseID, u32 battler, u32 ability, u32 special, u32
                 effect++;
             }
             break;
+        case ABILITY_TERA_SHIFT:
+                if (!gSpecialStatuses[battler].switchInAbilityDone &&
+                     gBattleMons[gBattlerAttacker].species == SPECIES_TERAPAGOS_NORMAL &&
```

Comment:
Should be battler instead of gBattlerAttacker.

Also this part has alignment issues and please use the `&&` operators on the next line to be consistent with the codebase.